source/adapters/level_zero/device.cpp

```diff
@@ -1107,10 +1107,15 @@ bool ur_device_handle_t_::useRelaxedAllocationLimits() {
 bool ur_device_handle_t_::useDriverInOrderLists() {
   // Use in-order lists implementation from L0 driver instead
   // of adapter's implementation.
-  static const bool UseDriverInOrderLists = [] {
+
+  ze_driver_handle_t ZeDriver = this->Platform->ZeDriver;
+
+  static const bool UseDriverInOrderLists = [&] {
     const char *UrRet = std::getenv("UR_L0_USE_DRIVER_INORDER_LISTS");
+    bool CompatibleDriver =
+        isDriverVersionNewerOrSimilar(ZeDriver, 1, 3, 29534);
```

Comment:
The two driver versions you used in this patch are not identical. Why?
If they are supposed to be, please create a `#define` or some `static const` variable that defines the correct version to support this feature.

Reply:
Yes these should be identical, since NEO had some bugs in the in-order implementation before this particular version. I'll update it to use a common #define then.